common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```
@@ -115,21 +115,24 @@ protected List<AccountRecord> getAccountsFilteredByInternal(@Nullable final Stri
         for (final AccountRecord account : allAccounts) {
             boolean matches = true;
```

Q: Does the matches variable have to be used here? would it be valid for the logic to be simplified to something like
```
if (mustMatchOnHomeAccountId && !StringUtil.equalsIgnoreCaseTrimBoth(homeAccountId, account.getHomeAccountId()) continue
... (same sort of logic for env and realm)
matchesAccounts.add(account);
```

---

Yeah, the problem is that there are too many conditions to check, which will make this form of condition matching really large.

---

But each condition block is independent right? If I were to expand upon what I was thinking:
```
if (mustMatchOnHomeAccountId && !StringUtil.equalsIgnoreCaseTrimBoth(homeAccountId, account.getHomeAccountId()) continue
if (mustMatchOnEnvironment && !StringUtil.equalsIgnoreCaseTrimBoth(environment, account.getEnvironment()) continue
if (mustMatchOnRealm && !StringUtil.equalsIgnoreCaseTrimBoth(realm, account.getRealm()) continue

matchesAccounts.add(account);
```
Since we're failing fast now, it doesn't really matter what the result of the previous if statement was (like if we got to the environment part, we know that home account id matched or account id just didn't matter), so there isn't really a need to carry the matches variable anymore, right?

---

Thanks for expanding! You're right. That is another way of simplifying this check.

---

Updated.